vignettes/adtte.Rmd

```diff
@@ -0,0 +1,265 @@
+
+This article describes creating an `ADTTE` (time-to-event) ADaM with common oncology
+endpoint parameters.
+
+The main part in programming a time-to-event dataset is the definition of the
+events and censoring times. `{admiral}`/`{admiralonco}` supports single events like
+death (Overall Survival) or composite events like disease progression or death (Progression
+Free Survival). More than one source dataset can be used for the definition of the event
+and censoring times.
+
+The majority of the functions used here exist from `{admiral}`, except for the
+`tte_sources` helper object, provided as an example from `{admiralonco}`.
+In practice, each company would create their own version of this, as likely the
+exact specifications such as filtering condition or description metadata will vary.
+
+**Note**: *All examples assume CDISC SDTM and/or ADaM format as input unless
+otherwise specified.*
+
+## Required Packages
+
+The examples of this vignette require the following packages.
+
+```{r, warning=FALSE, message=FALSE}
+library(admiral)
+library(admiralonco)
+library(dplyr)
+library(admiral.test)
+```
+
+```{r, warning=FALSE, message=FALSE, include=FALSE}
+library(lubridate)
+```
+
+# Programming Workflow
+
+* [Read in Data](#readdata)
+* [Derive Parameters (`CNSR`, `ADT`, `STARTDT`)](#parameters)
+* [Derive Analysis Value (`AVAL`)](#aval)
+* [Derive Analysis Sequence Number (`ASEQ`)](#aseq)
+* [Add ADSL Variables](#adslvars)
+
+## Read in Data {#readdata}
+
+To start, all datasets needed for the creation of the time-to-event dataset
+should be read into the environment. This will be a company specific process.
+
+For example purpose, the ADaM datasets---which are included
+in `{admiral}` and `{admiralonco}`---are used. An alternative might be to use
+`ADEVENT` as input.
+
+```{r}
+data("admiral_adsl")
+data("admiral_adrs")
+adsl <- admiral_adsl
+adrs <- admiral_adrs
+```
+
+## Derive Parameters (`CNSR`, `ADT`, `STARTDT`) {#parameters}
+
+To derive the parameter dependent variables like `CNSR`, `ADT`, `STARTDT`,
+`EVNTDESC`, `SRCDOM`, `PARAMCD`, ... the `admiral::derive_param_tte()` function can be
+used. It adds one parameter to the input dataset with one observation per
+subject. Usually it is called several times.
+
+For each subject it is determined if an event occurred. In the affirmative the
+analysis date `ADT` is set to the earliest event date. If no event occurred, the
+analysis date is set to the latest censoring date.
+
+The events and censorings are defined by the `admiral::event_source()` and the
+`admiral::censor_source()` class respectively. It defines
+
+- which observations (`filter` parameter) of a source dataset (`dataset_name`
+parameter) are potential events or censorings,
+- the value of the `CNSR` variable (`censor` parameter), and
+- which variable provides the date (`date` parameter).
+
+The date can be provided as date (`--DT` variable), datetime (`--DTM` variable),
+or character ISO-8601 date (`--DTC` variable).
```

Comment:
We decided to not accept `--DTC` variabls going forward so this will have to be updated.

Reply:
are you saying this is something to update now before the testing phase starts next month? or you're giving a heads up later in line with once admiral core makes this update? (as i copied this bit from admiral core vignette - so probably makes sense one person updates this across the packages)

p.s. i think the admiral core partial date imputation refactor i proposed on our Box will cause such impact in many places, so this will be one big update.

Reply:
I think we can keep it as it for now but will have to revisit once the changes to ate imputation have been made.

Reply:
agree! will resolve for now and we'll update this across all the admiral family of packages later